libcaja-private/caja-dnd.c

```diff
@@ -457,12 +459,31 @@ caja_drag_default_drop_action_for_icons (GdkDragContext *context,
     dropped_file = caja_file_get_existing_by_uri (dropped_uri);
     target_file = caja_file_get_existing_by_uri (target_uri_string);
 
+    if (eel_uri_is_desktop (dropped_uri))
```

Comment:
If `eel_uri_is_desktop(uri)` returned TRUE means that the uri starts with `x-caja-desktop:` prefix, for instance the **Computer** icon placed on user's desktop (x-caja-desktop:///computer)

Reply:
That code is related to the second expected behaviour.

- Desktop icon files cannot be dropped on other files or folders.

With desktop icon files I mean any CajaDesktopIconFile, as described in caja-desktop-icon-file.h. My intention was to stop the user from performing dnd operations that lead to a sure error and leave the dnd animation as the only feedback. I think that improves the user experience, but I don't know if it's a majority opinion.

Reply:
IMHO it should be handled by gvfs backends. `caja_drag_default_drop_action_for_icons` sets the default dnd action, when it's available, and it doesn't stop/end a dnd job by setting `*action = 0;`